### PR TITLE
[move source language] Cleanup loc's after rework

### PR DESCRIPTION
**Pull request by tnowacki**
## Motivation

- Fixed loc reporting in edge cases
- Fixed loc reporting for field mutate

## Test Plan

- ran em

### CODE REVIEW

File: language/move-lang/tests/move_check/typing/module_call.exp [hunk 1]
```diff
@@ -41,8 +41,8 @@ error:
  43 │         let () = X::bing(X::baz(X::bar(X::foo())));
     │                          ^^^^^^^^^^^^^^^^^^^^^^^^ Invalid type argument
```
Comment (bob-wilson) [inline language/move-lang/tests/move_check/typing/module_call.exp]: So.... the changes here are an improvement, but I'm still having a hard time understanding these errors. The baz function takes 2 generic type arguments that are inferred by the compiler (in some cases). It is confusing to tell the user that they have an invalid type argument when they haven't specified any type arguments. Would it make sense to say something like "Inference of type argument failed"? It's especially confusing here because there are 2 type arguments expected but the message says that it is expected a single type (presumably for the first type argument). Any ideas how to make that better?

Comment (tnowacki) [reply language/move-lang/tests/move_check/typing/module_call.exp]: Type inference really didn't fail here. The function takes a two arguments, but it was passed an expression list instead of two explicit arguments. In other words, there should be a splat here (but I haven't added that to the language)

The thing I would point out, is that this error message actually comes after the invalid argument error. So hopefully a user would read that one first? The only other thing I could think of doing is adding what type parameter it is... which isn't great....

Comment (bob-wilson) [reply language/move-lang/tests/move_check/typing/module_call.exp]: Yeah, I understand the problem. I just think we should find a way to avoid the term "type argument" here since I can't see any type arguments in that code. It's possible that is because I don't know how you use that term, but I would expect to get that error if I wrote something like:

Comment (tnowacki) [reply language/move-lang/tests/move_check/typing/module_call.exp]: Yeah, I think pointing to what type parameter it is might help. but its a hard thing to do in the current setup
